flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/source/DataIterator.java

```diff
@@ -62,18 +96,25 @@ public boolean hasNext() {
   @Override
   public T next() {
     updateCurrentIterator();
+    recordOffset += 1;
     return currentIterator.next();
   }
 
+  public boolean isCurrentFileDone() {
```

Comment:
I think there's probably a better name for this, like `currentFileHasNext` that fits better with the `Iterator` naming.

Reply:
`currentFileHasNext` sounds good to me.